### PR TITLE
#9 Add missing properties

### PR DESCRIPTION
**Pull request by oubchid**

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -70,12 +70,14 @@ export interface IEmailProperties {
...
   to: IEmailAddress[] | null;
   subject: string;
+  date: Date;
   size: number;
   preview: string;
   attachments: Attachment[] | null;
   createdModSeq: number;
   updatedModSeq: number;
-  receivedAt: IUtcDate;
+  receivedAt: IUtcDate; // ? receivedAt
```
Comment (alagane) [inline src/types.ts:79]: https://jmap.io/spec-mail.html#properties-of-the-email-object

**Review by fabienmoyon — DISMISSED** [review]

File: src/types.ts [hunk 2]
```diff
@@ -140,27 +198,20 @@ export interface IEmailAddress {
...
+export interface IThreadProperties {
   id: string;
-  emails: IThreadEmail[];
-  createdModSeq: number;
-  updatedModSeq: number;
-  deleted: Date | null;
+  emailsIds: string[];
 }
 
 export interface IMailbox extends IMailboxProperties, IMailboxData {}
```
Comment (alagane) [inline src/types.ts:209]: IMailboxData seems unused, it comes from https://jmap.io/server.html#3-mailboxes and is outdated.

File: src/types.ts [hunk 3]
```diff
@@ -251,44 +308,38 @@ export interface ISetError {
...
-  highModSeq: number;
-  highModSeqEmail: number;
-  highModSeqThread: number;
-  highModSeqMailbox: number;
-  lowModSeqEmail: number;
-  lowModSeqThread: number;
-  lowModSeqMailbox: number;
-}
-
 export interface IMessage {
```
Comment (alagane) [inline src/types.ts:343]: This one seems unused too.

File: src/types.ts [hunk 4]
```diff
@@ -1,13 +1,17 @@
...
-export type IEntityProperties = IEmailProperties | IMailboxProperties;
+export type IEntityProperties = IMailboxProperties | IEmailProperties;
 
-export type IFilterCondition = IEmailFilterCondition;
+/**
+ * See https://jmap.io/spec-core.html#query
+ */
+export type IFilterCondition = IMailboxFilterCondition | IEmailFilterCondition;
 
 export type IArguments = IGetArguments<IEntityProperties> | IQueryArguments<IEmailFilterCondition>;
```
Comment (alagane) [inline src/types.ts:16]: Missing `ISetArguments<IEntityProperties>`

File: src/types.ts [hunk 5]
```diff
@@ -17,30 +21,123 @@ export interface IReplaceableAccountId {
...
+  hasMoreChanges: boolean;
+  created: string[];
+  updated: string[];
+  destroyed: string[];
+}
+
+/**
+ * See https://jmap.io/spec-core.html#set
+ */
 export interface ISetArguments<Foo> extends IReplaceableAccountId {
```
Comment (alagane) [inline src/types.ts:59]: You could use extends like for `IGetArguments<Properties extends IEntityProperties>`, to be more explicit about the type of Foo.

**Review by alagane — APPROVED** [review]

**Review by fabienmoyon — APPROVED** [review]

**Pull request merged**